pkg/nuclide-remote-connection/lib/RemoteFile.js

```diff
@@ -254,10 +254,11 @@ export class RemoteFile {
   }
 
   async read(flushCache?: boolean): Promise<string> {
-    const data = await this._getFileSystemService().readFile(this._localPath);
+    const encoding = this.getEncoding();
+    const options = encoding ? {encoding} : {};
+    const data = await this._getFileSystemService().readFile(this._localPath, options);
     const contents = data.toString();
```

Comment:
Something about this feels off. Does this work? Doesn't this implicitly encode the buffer as a UTF-8 string? Or does this work because the `readFile` call implicitly decodes into a UTF-8 string?

Is that guaranteed to work? IMO it would be clearer and more preferable to keep the `readFile` call encoding-agnostic, let it keep returning a raw Buffer, and do the decoding here (as with Atom's implementation).